calendar-bundle/src/Migration/OverwriteMetaMigration.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\Migration;
+
+use Contao\CoreBundle\Migration\Version505\OverwriteMetaMigration as BaseMigration;
```

Comment:
The base class is marked `@internal`, so technically we cannot use it here. Maybe we should provide a non-internal base class that can be used by everybody?

Reply:
> The base class is marked `@internal`, so technically we cannot use it here.

Because we’re using `self.version` as our `core-bundle` requirement using `@internal` features in the other bundles should never be a problem.

Semantically I would interpret `@internal` on the root namespace level here, so every subnamespace of `\Contao` would be allowed to access/use it.

Reply:
> Maybe we should provide a non-internal base class that can be used by everybody?

I don’t think that is nessecary.

Reply:
So you think no extension but the default bundles uses the metadata overwrite mechanism?

Reply:
They would have to use it in the exact same way. Our migration is too specific to be an abstract one I think. We skip the check for `varchar` to `text` for `imageUrl` for example because this is already a `text` column in `tl_content`. The Column names are fixed and are not configurable, etc..
If someone needs a similar migration, the can just copy it and adapt it to their needs.